### ISSUE TITLE
ENet-Label-Torch is available now (a light-weight and effective lane detection model)

### ISSUE THREAD

**Issue by cardwing** [amusi/awesome-lane-detection#2]
Our **ENet-Label-Torch** has been released. More details can be found in [my repo](https://github.com/cardwing/Codes-for-Lane-Detection).

Key features:

(1) ENet-label is a **light-weight** lane detection model based on [ENet](https://arxiv.org/abs/1606.02147) and adopts **self attention distillation** (more details can be found in our paper which will be published soon).

(2) It has **20** × fewer parameters and runs **10** × faster compared to the state-of-the-art SCNN, and achieves **72.0** (F1-measure) on CULane testing set (better than SCNN which achieves 71.6).

(**Do not hesitate to try our model!!!**)

Performance on CULane testing set (F1-measure):

|Category|[SCNN-Torch](https://github.com/XingangPan/SCNN)|SCNN-Tensorflow|[ENet-Label-Torch](https://github.com/cardwing/Codes-for-Lane-Detection)|
|:---:|:---:|:---:|:---:|
|Normal|90.6|90.2|**90.7**|
|Crowded|69.7|71.9|70.8|
|Night|66.1|64.6|65.9|
|No line|43.4|45.8|44.7|
|Shadow|66.9|73.8|70.6|
|Arrow|84.1|83.8|**85.8**|
|Dazzle light|58.5|59.5|**64.4**|
|Curve|64.4|63.4|**65.4**|
|Crossroad|1990|4137|2729|
|Total|71.6|71.3|**72.0**|
|Runtime(ms)|133.5|--|**13.4**|
|Parameter(M)|20.72|--|**0.98**|

**Comment by mengzhibin** [comment]
Hi, We do not hesitate to try your model, But lua and torch will stop many people. That will be great if there is pytorch and python version. All in all, thank you again.

**Comment by MaybeShewill-CV** [comment]
@mengzhibin @cardwing Things will greater if there is tensor flow version:)

**Comment by cardwing** [comment]
I will implement both Pytorch version and TensorFlow version when I am free. However, you can just read the torch codes, then write the PyTorch / TensorFlow codes and pull request in my Github repo. I will respond to you soon.

**Comment by harryhan618** [comment]
@mengzhibin Hi! May I recommend my repo? Both SCNN and LaneNet are state of art lane detection cnn model. I implement them [here](https://github.com/harryhan618/SCNN_Pytorch) and [here](https://github.com/harryhan618/LaneNet) in pytorch. Hope this helps. Thanks!

**Comment by mengzhibin** [comment]
Thanks for your work, that's will help a lot. @harryhan618

**Comment by cardwing** [comment]
@mengzhibin [ERFNet-CULane-PyTorch](https://github.com/cardwing/Codes-for-Lane-Detection) has been released. (It can achieve 72.3 F1-measure in CULane testing set)

**Comment by mengzhibin** [comment]
@cardwing Thanks for your work of lane detection. I have checked the code.

**Comment by cardwing** [comment]
@mengzhibin You are welcome. And you can add other features, e.g., SAD, to the PyTorch codes. You can obtain better performance in all benchmarks.
